### PR TITLE
Improved sorting and error messages for cfbs.json files

### PR DESCRIPTION
**Pull request by olehermanse**
- Moved validate_config_for_build_field to validate.py
- Added build validation to validate command
- Adapted CFBSConfig and CFBSJson to work with alternate paths
- validate.py: Adapted the main function to be aligned with the validate command
- Moved top_level_keys and module_keys to be globals for reuse
- Fixed cfbs.json sorting by adding new keys
- Fixed sorting for "provides" field
- Adapted JSON sorting to allow passing "alphabetic" or a list of strings to sort by
- Replaced lambda functions to make the cfbs sorting rules a bit more readable
- Made cfbs pretty also sort module entries inside "build" array
- Added more tests for pretty printing

### CODE REVIEW

File: cfbs/commands.py [hunk 1]
```diff
@@ -98,34 +97,28 @@ def pretty_command(filenames: list, check: bool, keep_order: bool) -> int:
...
-            "description",
-            "tags",
-            "repo",
-            "by",
-            "version",
-            "commit",
-            "subdirectory",
-            "dependencies",
-            "steps",
+        # These sorting rules acchieve 3 things:
```
Comment (olehermanse) [inline cfbs/commands.py:100]:
```suggestion
        # These sorting rules achieve 3 things:
```

File: cfbs/commands.py [hunk 2]
```diff
@@ -98,34 +97,28 @@ def pretty_command(filenames: list, check: bool, keep_order: bool) -> int:
...
-            "description",
-            "tags",
-            "repo",
-            "by",
-            "version",
-            "commit",
-            "subdirectory",
-            "dependencies",
-            "steps",
+        # These sorting rules achieve 3 things:
```
Comment (larsewi) [inline cfbs/commands.py:100]: You could do an interactive rebase and squash the typo-fix with the commit that introduced it, hiding the fact that you are only human 😉

**Review by larsewi — APPROVED** [review]
Nice 🚀

File: cfbs/commands.py (hunk 2 above)
Comment (olehermanse) [reply cfbs/commands.py:100]: Done ;)

File: cfbs/cfbs_json.py [hunk 3]
```diff
@@ -4,6 +4,27 @@
 from cfbs.utils import read_json, user_error
 from cfbs.pretty import pretty
 
+# Globals for the keys in cfbs.json and their order
+# Used for validation and prettifying / sorting.
+# TODO: Update these lists with the new keys
```
[the remark this answers is not in the document]
Comment (olehermanse) [reply cfbs/cfbs_json.py]: Done :)

**Pull request merged**